lib/NormalModuleFactory.js

```diff
@@ -474,18 +478,23 @@ class NormalModuleFactory extends ModuleFactory {
 					return continueCallback();
 				}
 
-				normalResolver.resolve(
-					contextInfo,
-					context,
-					unresolvedResource,
-					resolveContext,
-					(err, resolvedResource, resolvedResourceResolveData) => {
-						if (err) return continueCallback(err);
-						resource = resolvedResource;
-						resourceResolveData = resolvedResourceResolveData;
-						continueCallback();
-					}
-				);
+				if (isDataURI(unresolvedResource)) {
```

Comment:
This matches `unresolvedResource` twice against the data url regexp. Here and for `getMimetype`.

It should be possible to only use `getMimetype` and check `if(mimetype)` here.

Maybe rename to make it clearer: `mimetype` -> `dataUriMimetype`

Reply:
fixed